crates/sui-node/src/lib.rs

```diff
@@ -264,6 +269,14 @@ impl SuiNode {
         Ok(node)
     }
 
+    pub async fn subscribe_to_epoch_change(&self) -> tokio::sync::broadcast::Receiver<Committee> {
+        self.reconfig_channel.lock().await.resubscribe()
```

Comment:
Does this need to be synchronized? If not, we can avoid waiting to acquire the lock and instead call `self._checkpoint_executor_handle.subscribe_to_end_of_epoch()`

Reply:
this is wrong and will deadlock. 
Fixed coming in next push by using `self.checkpoint_executor_handle.subscribe_to_end_of_epoch()`